### ISSUE TITLE
HiDPI causes SVG rendering of cat packs to render pixelated

### ISSUE THREAD

**Issue by espartia** [PrismLauncher/PrismLauncher#3185]
### Operating System

Linux

### Version of Prism Launcher

Prism Launcher 9.1

### Version of Qt

Qt 6.7.2-r5 (Gentoo)

### Description of bug

When using an SVG for a cat pack on a HiDPI display, it becomes pixelated as if it was a raster image.

### Steps to reproduce

* Set the display scaling higher than 1.0
* Add an SVG file into the cat packs folder
* Select that SVG as the cat pack
* Press the cat button, if it isn't already enabled

### Suspected cause

Treating vector images as raster images, and having QT render them into raster images, then scaling them to the correct DPI, rather than scaling them to the correct DPI before it is converted to raster.

### This issue is unique

- [X] I have searched the issue tracker and did not find an issue describing my bug.

**Comment by espartia** [comment]
I'm using X11, btw, but I suspect the issue has nothing to do with the graphics stack itself (besides QT's rendering of the image)

**Comment by TheKodeToad** [comment]
Indeed, this also happens on Windows

**Comment by Trial97** [comment]
Does this and https://github.com/PrismLauncher/PrismLauncher/issues/2195 share the same root cause?

**Comment by espartia** [comment]
I don't think so. They seem to be unrelated problems - The SVG is actually pixelated and not blurry.
This is how my custom SVG image is rendering, for reference:
![image](https://github.com/user-attachments/assets/53222d9c-1054-4323-96e2-2da109f1b8f0)

**Comment by espartia** [comment]
This is specific to SVGs, and the behavior makes sense for raster images such as PNGs and JPEGs

**Comment by TheKodeToad** [comment]
> This is specific to SVGs, and the behavior makes sense for raster images such as PNGs and JPEGs

Not really, pretty sure they're meant to be interpolated - if you don't do that they can be distorted

**Comment by espartia** [comment]
This could probably be fixed for SVGs by rendering SVGs at the correct resolution, taking into account DPI, before rendering them. I'll see if I can make a PR.

**Comment by espartia** [comment]
It seems a bit more complicated - I think I'll need QtSvg to render SVGs properly, with the correct size.

**Comment by TayouVR** [comment]
this was mentioned in #development on discord a while back and I tried to fix it, but couldn't do it...
went through a few things, my last attempt was using `this->viewport()->devicePixelRatio()` around InstanceView.cpp L462, but that didn't work

**Comment by LUIGI-COM** [comment]
I’m experiencing a similar issue, but with PNG and JPG images.
I'm on Linux using Wayland and running PrismLauncher version 9.4 from the Arch extra repo. The images in the catpacks are high resolution and appear perfectly sharp at 100% display scaling. However, when I increase the scaling to 125%, all catpack images become pixelated.

**Comment by severtheskyline** [comment]
Yeah, I'm having the same issue but with pngs as well:

<img width="1557" height="997" alt="Image" src="https://github.com/user-attachments/assets/13830c60-8a39-497c-aaa4-8efbb25a771c" />

Considering that this is the original image:

<details>
  <summary>Click to Teto</summary>

<img width="2351" height="5000" alt="Image" src="https://github.com/user-attachments/assets/733b2a52-220f-49b6-a241-231c8f0b74cd" />
</details>

I'd say that is is a really bad conversion.
